src/test/java/seedu/duke/models/schema/ScheduleTest.java

```diff
@@ -70,4 +73,115 @@ void addModuleTest_invalidTargetSem_expectException() {
+        Schedule schedule = new Schedule();
+        schedule.addModule("CS1010", 1);
+        schedule.addModule("CS2040C", 2);
+        schedule.deleteModule("CS2040C");
+        ArrayList<String> testArray= new ArrayList<>();
+        testArray.add("CS1010");
+        assertEquals(testArray, schedule.getModulesPlanned().getModuleCodes());
+
+    }
+
+    //failure scenario 1: String containing invalid module
+    // --> throw MissingModuleException
+    @Test
+    void deleteModuleTest_invalidModule_expectException() {
+        Schedule schedule = new Schedule("CS1231S MA1511", new int[]{2, 0, 0, 0, 0, 0, 0, 0});
+        assertThrows(MissingModuleException.class, () -> schedule.deleteModule("wrong"));
+
+    }
+
+    //failure scenario 2: String containing valid module, module not in schedule
+    //--> throws MissingModuleException
+    @Test
+    void deleteModuleTest_missingModule_expectException() {
+        Schedule schedule = new Schedule("CS1231S MA1511", new int[]{2, 0, 0, 0, 0, 0, 0, 0});
+        assertThrows(MissingModuleException.class, () -> schedule.deleteModule("CS2040C"));
+    }
+
+    //failure scenario 3: String containing valid module, module in schedule
+    //module is prerequisite of modules ahead--> throws MandatoryPrereqException
+    @Test
+    void deleteModuleTest_mandatoryPrereq_expectException() {
+        Schedule schedule = new Schedule("CS1010 CS2040C", new int[]{1, 1, 0, 0, 0, 0, 0, 0});
+        assertThrows(MandatoryPrereqException.class, () -> schedule.deleteModule("CS1010"));
+    }
+
+    //success scenario 1: String containing valid module, module is in schedule, int containing valid targetSem
+    //shifting earlier does not affect other modules--> module shifted earlier
+    @Test
+    void shiftModuleTest_correctInputsShiftEarlier_expectModuleShifted() throws Exception {
+        Schedule schedule = new Schedule();
+        schedule.addModule("CS1010", 3);
+        schedule.addModule("MA1511", 2);
+        schedule.shiftModule("CS1010", 1);
+        ArrayList<String> testArray= new ArrayList<>();
+        testArray.add("CS1010");
+        testArray.add("MA1511");
+        assertEquals(testArray, schedule.getModulesPlanned().getModuleCodes());
+
+    }
+
+    //success scenario 2: String containing valid module, module is in schedule, int containing valid targetSem
+    //shifting later does not affect other modules--> module shifted later
+    @Test
+    void shiftModuleTest_correctInputsShiftLater_expectModuleShifted() throws Exception {
+        Schedule schedule = new Schedule();
+        schedule.addModule("CS1010", 1);
+        schedule.addModule("MA1511", 2);
+        schedule.shiftModule("CS1010", 3);
+        ArrayList<String> testArray= new ArrayList<>();
+        testArray.add("MA1511");
+        testArray.add("CS1010");
+        assertEquals(testArray, schedule.getModulesPlanned().getModuleCodes());
+
+    }
```

Comment:
i think it would be good to name testArray as expectedArray to clearly indicate which array is the expected result. I would suggest adding lines between the expected array and the array from shiftModule, for readability!